core/src/main/java/org/infinispan/encoding/DataConversion.java

```diff
@@ -43,152 +38,155 @@
-      return new DataConversion(this.encoderClass, wrapperClass, this.requestMediaType, this.storageMediaType,
-            this.isKey);
+      return new DataConversion(this.encoderClass, wrapperClass, this.requestMediaType, this.isKey);
    }
 
+   /**
+    * @deprecated Since 11.0, will be removed with no replacement
+    */
+   @Deprecated
    public void overrideWrapper(Class<? extends Wrapper> newWrapper, ComponentRegistry cr) {
-      this.wrapper = null;
+      this.customWrapper = null;
       this.wrapperClass = newWrapper;
       cr.wireDependencies(this);
    }
 
    /**
-    * Obtain the configured {@link MediaType} for this instance, or assume sensible defaults.
+    * @deprecated Since 11.0. To be removed in 14.0, with no replacement.
     */
-   private MediaType getStorageMediaType(Configuration configuration, boolean embeddedMode, boolean internalCache, PersistenceMarshaller persistenceMarshaller) {
-      EncodingConfiguration encodingConfiguration = configuration.encoding();
-      ContentTypeConfiguration contentTypeConfiguration = isKey ? encodingConfiguration.keyDataType() : encodingConfiguration.valueDataType();
-      Marshaller userMarshaller = persistenceMarshaller.getUserMarshaller();
-      MediaType mediaType = userMarshaller.mediaType();
-      boolean heap = configuration.memory().storageType() == StorageType.OBJECT;
-      // If explicitly configured, use the value provided
-      if (contentTypeConfiguration.isMediaTypeChanged()) {
-         return contentTypeConfiguration.mediaType();
-      }
-      // Indexed caches started by the server will assume application/protostream as storage media type
-      if (!embeddedMode && configuration.indexing().enabled() && contentTypeConfiguration.mediaType() == null) {
-         return MediaType.APPLICATION_PROTOSTREAM;
-      }
-      if (internalCache) return MediaType.APPLICATION_OBJECT;
-
-      if (embeddedMode) {
-         return heap ? MediaType.APPLICATION_OBJECT : mediaType;
-      }
-
-      return MediaType.APPLICATION_UNKNOWN;
-   }
-
+   @Deprecated
    public boolean isConversionSupported(MediaType mediaType) {
-      return storageMediaType == null || encoderRegistry.isConversionSupported(storageMediaType, mediaType);
+      if (encoderRegistry == null) return false;
+      return encoderRegistry.isConversionSupported(storageMediaType, mediaType);
    }
 
+   /**
+    * @deprecated Since 11.0. To be removed in 14.0, with no replacement.
+    */
+   @Deprecated
    public Object convert(Object o, MediaType from, MediaType to) {
-      if (o == null) return null;
-      if (encoderRegistry == null) return o;
-      Transcoder transcoder = encoderRegistry.getTranscoder(from, to);
-      return transcoder.transcode(o, from, to);
+      return encoderRegistry.convert(o, from, to);
    }
 
+   /**
+    * @deprecated Since 11.0. To be removed in 14.0, with no replacement.
+    */
+   @Deprecated
```

Comment:
I just updated #8196, could you handle it first, since you are including that commit in this PR?